sdk/cosmosdb/cosmos/src/client/ChangeFeed/changeFeedUtils.ts

```diff
@@ -0,0 +1,135 @@
+import { Constants } from "../../common";
+import { ChangeFeedStartFromTime } from "./ChangeFeedStartFromTime";
+import { QueryRange } from "../../routing";
+import { FeedRangeInternal } from "./FeedRange";
+
+/**
+ * @hidden
+ * Validates the change feed options passed by the user
+ */
+export function validateChangeFeedIteratorOptions(options: ChangeFeedIteratorOptions): void {
+  if (!isChangeFeedIteratorOptions(options)) {
+    throw new ErrorResponse("Invalid Changefeed Iterator Options.");
+  }
+  if (options?.maxItemCount && typeof options?.maxItemCount !== "number") {
+    throw new ErrorResponse("maxItemCount must be number");
+  }
+  if (options?.maxItemCount !== undefined && options?.maxItemCount < 1) {
+    throw new ErrorResponse("maxItemCount must be a positive number");
+  }
+}
+
+function isChangeFeedIteratorOptions(options: unknown): options is ChangeFeedIteratorOptions {
+  if (Object.keys(options).length === 0) {
+    return true;
+  }
+  return options && !(isPrimitivePartitionKeyValue(options) || Array.isArray(options));
+}
+
+/**
+ * @hidden
+ * Checks if pkRange entirely covers the given overLapping range or there is only partial overlap.
+ *
+ * If no complete overlap, exact range which overlaps is retured which is used to set minEpk and maxEpk headers while quering change feed.
+ */
+export async function checkEpkHeaders(
+  epkRange: QueryRange,
+  overLappingRange: PartitionKeyRange
+): Promise<[string, string]> {
+  if (
+    overLappingRange.minInclusive >= epkRange.min &&
+    overLappingRange.maxExclusive <= epkRange.max
+  ) {
+    return [undefined, undefined];
+  } else if (
+    overLappingRange.minInclusive <= epkRange.min &&
+    overLappingRange.maxExclusive >= epkRange.max
+  ) {
+    return [epkRange.min, epkRange.max];
+  }
+  // Right Side of overlapping range is covered
+  else if (
+    overLappingRange.minInclusive <= epkRange.min &&
+    overLappingRange.maxExclusive <= epkRange.max &&
+    overLappingRange.maxExclusive >= epkRange.min
+  ) {
+    return [epkRange.min, overLappingRange.maxExclusive];
+  }
+  // Left Side of overlapping range is covered
+  else {
+    return [overLappingRange.minInclusive, epkRange.max];
+  }
+}
+
+/**
+ * @hidden
+ * Checks if the object is a valid EpkRange
+ */
+/* eslint-disable @typescript-eslint/explicit-module-boundary-types */
+export function isEpkRange(obj: any): boolean {
+  return (
+    obj instanceof FeedRangeInternal &&
+    typeof obj.minInclusive === "string" &&
+    typeof obj.maxExclusive === "string" &&
+    obj.minInclusive >=
+      Constants.EffectivePartitionKeyConstants.MinimumInclusiveEffectivePartitionKey &&
+    obj.maxExclusive <=
+      Constants.EffectivePartitionKeyConstants.MaximumExclusiveEffectivePartitionKey &&
+    obj.maxExclusive > obj.minInclusive
+  );
+}
+
+/**
+ * @hidden
+ * Verify validity of partition key.
+ */
+export function isPartitionKey(partitionKey: unknown) {
```

Comment:
Please move this function to src\utils\typeChecks.ts

Reply:
done